src/v/cluster/controller_backend.cc

```diff
@@ -484,12 +530,47 @@ ss::future<std::error_code> controller_backend::process_partition_update(
         co_return std::error_code(errc::waiting_for_recovery);
     }
 
+    /**
+     * Check if this NTP was moved from other shard on current broker, if so we
+     * have to wait for it to be shutdown. Partition creation will be dispatched
+     * to target partition manager from the core where the partition is
+     * currently present. On this core we will just wait for partition to be
+     * created.
+     */
+    if (has_node_local_replicas(_self, previous.replicas)) {
```

Comment:
Don’t we have to take a multi-core lock ? Source and destination 

I am missing the details on how we fail the TCP conns writing to this machine. To me this has to be wired all the way up to the partition. 

With only 3 mins of thinking the protocol should be like: 

1. Shutdown partition, and have a Boolean flag that says not-ready or smth so we fail partitions
2. We then update the destination core 
3. We then update the lookup data structures. 

I need help understanding how does the TCP requests fail during the movement of cores.

Reply:
Or maybe the original core will route traffic to the destination core during the transition.

Reply:
I think I will write some more detailed description about it. In general requests will start failing as soon as ntp will be removed from `shard_table`.